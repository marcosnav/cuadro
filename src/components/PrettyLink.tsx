import React, { FC } from 'react';
import styled from 'styled-components';

const Link = styled.a`
  color: ${(props) => props.theme.TEXT};
  font-weight: bold;
  text-decoration: underline;
  transition: opacity 0.3s ease;

  &:hover {
    opacity: 0.4;
  }
`;

interface IProps {
  href?: string;
  onClick?: () => void;
}

const PrettyLink: FC<IProps> = ({ href, onClick, children }) => {
  const action = (ev: React.MouseEvent) => {
    if (onClick) {
      ev.preventDefault();
      onClick();
    }
  };

  return (
    <Link href={href} target='_blank' onClick={action}>
      {children}
    </Link>
  );
};

export default PrettyLink;
