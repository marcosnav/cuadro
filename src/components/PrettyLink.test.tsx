import { mount } from 'enzyme';
import 'jest-enzyme';
import 'jest-styled-components';
import React from 'react';
import PrettyLink from './PrettyLink';

describe('<PrettyLink/>', () => {
  test('renders properly', () => {
    const clickSpy = jest.fn();
    const component = mount(<PrettyLink onClick={clickSpy}>link text</PrettyLink>);
    component.simulate('click');
    expect(component).toMatchSnapshot();
    expect(clickSpy).toHaveBeenCalled();
  });
});
