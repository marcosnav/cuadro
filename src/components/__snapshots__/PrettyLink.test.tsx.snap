// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrettyLink/> renders properly 1`] = `
.c0 {
  font-weight: bold;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
}

.c0:hover {
  opacity: 0.4;
}

<PrettyLink
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <styled.a
    onClick={[Function]}
    target="_blank"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  font-weight: bold;
  text-decoration: underline;
  transition: opacity 0.3s ease;

  &:hover {
    opacity: 0.4;
  }
",
            ],
          },
          "displayName": "styled.a",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "a",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      target="_blank"
    >
      <a
        className="c0"
        onClick={[Function]}
        target="_blank"
      >
        link text
      </a>
    </StyledComponent>
  </styled.a>
</PrettyLink>
`;
